charts/prometheus-pushgateway/templates/deployment.yaml

```diff
@@ -71,13 +73,21 @@ spec:
       affinity:
 {{ toYaml .Values.affinity | indent 8 }}
     {{- end }}
-      {{- if .Values.persistentVolume.enabled }}
+      {{- if or .Values.persistentVolume.enabled .Values.secrets }}
     {{- if .Values.securityContext }}
       securityContext:
 {{ toYaml .Values.securityContext | indent 8 }}
     {{- end }}
       volumes:
+        {{- range .Values.secrets }}      
```

Comment:
Should be named "extraVolumes" and have an example in the values file too.

Reply:
Hmmmm, imho, I would like to keep it this way as  mounting a secret is different from mounting other kind of storage. Unless we drop the idea of letting the user to add secret and let him add whatever needed as a yaml value. We could even add the two options actually (secret or a raw yaml value) What do you think @monotek ?

Reply:
But with this function you could add genric volumes and therefore i'd like to see a generic name there.

Would be also mor consistent with prometheus chart. See: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L650-L687

Reply:
There you go :-)